lib/timer.py

```diff
@@ -112,20 +124,30 @@ def run_jobs(cfg: UAConfig, current_time: datetime):
     Persist jobs-status with calculated next_run values to aid in timer
     state introspection for jobs which have not yet run.
     """
-    jobs_status = cfg.read_cache("jobs-status") or {}
+    jobs_status_obj = timer_jobs_state_file.read()
+
+    if jobs_status_obj is None:
+        # We do this for the first run of the timer job, where the file
+        # doesn't exist
+        jobs_status_obj = AllTimerJobsState(
+            metering=None, update_messaging=None
+        )
+
     for job in UACLIENT_JOBS:
-        if job.name in jobs_status:
-            next_run = jobs_status[job.name]["next_run"]
-            if next_run > current_time:
+        if getattr(jobs_status_obj, job.name):
+            next_run = getattr(jobs_status_obj, job.name).next_run
```

Comment:
by using `getattr` here (and `setattr` later), mypy can't check most of this for errors. But also I understand why you're doing it here.

If you can think of a way to avoid `getattr` then let's try it, but if not then that's okay too.

Reply:
@orndorffgrant I have modified the PR to make it more type friendly. Let me know what you think